ViAn/Filehandler/filehandler.cpp

```diff
@@ -2,225 +2,223 @@
 
 /**
  * @brief FileHandler::FileHandler
+ * Initilize filehandler and set
+ * default value for workspace
  */
 FileHandler::FileHandler() {
     this->project_id = 0; // zero out counter ids
     this->file_id = 0;
     this->dir_id = 0;
-    this->last_error = 0;
+    this->last_error = false;
     #ifdef _WIN32
-        this->work_space = "C:/";
+        this->work_space = create_directory("C:/");
     #elif __APPLE__
-        this->work_space = "/Applications/";
+        this->work_space =  create_directory("/Applications/");
     #elif __unix__
-        this->work_space = "~/";
+        this->work_space =  create_directory("~/");
     #endif
 
     //ID id = add_file("ViAn_config.txt"); Will be used to store current workspace and other run-to-run coonstans
 }
+
 /**
  * @todo save workspace to file
  * @brief FileHandler::set_workspace
- * @param newWorkSpace
+ * @param new_work_space
+ * Change default work_space, ie where project files are saved.
  */
-void FileHandler::set_workspace(std::string new_work_space){
-    this->work_space = new_work_space;
+void FileHandler::set_work_space(std::string new_work_space){
+    //this->work_space = new_work_space;
     //save_workspace();
 }
 
 /**
+ * @brief FileHandler::get_work_space
+ * @return default work_space
+ */
+QDir FileHandler::get_work_space()
+{
+    return this->get_dir(this->work_space);
+}
+
+/**
  * @brief FileHandler::create_project
  * creates project and associated files.
  * @param std::string name
  * @return Project* created project
  */
-Project* FileHandler::create_project(std::string proj_name){
-    Project* proj =  new Project(this->project_id, proj_name);
-    this->projects.insert(std::make_pair(proj->id, proj));
-    save_project(proj);
-    this->project_id++;
+Project* FileHandler::create_project(QString proj_name, std::string dir_path){
+    Project* proj =  new Project(this->project_id, proj_name.toStdString());   
+    ID root_dir;
+    if(dir_path != "")                          //Directory name provided
+        root_dir = create_directory(QString::fromStdString(dir_path));
+    else if(dir_path != "" && proj->dir == -1)  // No directory name provided, project has no directory.
+        root_dir = this->work_space;                 // Default save location to workspace
+    else if(dir_path == "" && proj->dir != -1)  // No Directory provided and project previosuly saved
+        root_dir = proj->dir;                        // Use present save location
+    else
+        root_dir = this->work_space;
+
+
+    proj->dir = create_directory(get_dir(root_dir).absoluteFilePath(QString::fromStdString(proj->name)));
+    add_project(proj);                          // Add project to file sytstem
+    save_project(proj);                         // Save project file
     return proj;
 }
 
```

Comment:
I have not! The if was a bit wrong so it didnt work. its fixed now tho. It's more flexible than it needs to be atm. As this will be implemented later (to choose directory on GUI level)